engine/src/main/java/org/pentaho/di/job/entries/ftpdelete/JobEntryFTPDelete.java

```diff
@@ -0,0 +1,1248 @@
+              filelist[i] = dirEntry.filename;
+              i++;
+            }
+          }
+        }
+      }
+
+      if ( isDetailed() ) {
+        logDetailed( "JobEntryFTPDelete.FoundNFiles", String.valueOf( filelist.length ) );
+      }
+      int found = filelist == null ? 0 : filelist.length;
+      if ( found == 0 ) {
+        result.setResult( true );
+        return result;
+      }
+
+      Pattern pattern = null;
+      if ( copyprevious ) {
+        // Copy the input row to the (command line) arguments
+        for ( int iteration = 0; iteration < rows.size(); iteration++ ) {
+          resultRow = rows.get( iteration );
+
+          // Get file names
+          String file_previous = resultRow.getString( 0, null );
+          if ( !Utils.isEmpty( file_previous ) ) {
+            list_previous_files.add( file_previous );
+          }
+        }
+      } else {
+        if ( !Utils.isEmpty( wildcard ) ) {
+          String realWildcard = environmentSubstitute( wildcard );
+          pattern = Pattern.compile( realWildcard );
+
+        }
+      }
+
+      if ( !getSuccessCondition().equals( SUCCESS_IF_ALL_FILES_DOWNLOADED ) ) {
+        limitFiles = Const.toInt( environmentSubstitute( getLimitSuccess() ), 10 );
+      }
+
+      // Get the files in the list...
+      for ( int i = 0; i < filelist.length && !parentJob.isStopped(); i++ ) {
+        if ( successConditionBroken ) {
+          throw new Exception( BaseMessages.getString( PKG, "JobEntryFTPDelete.SuccesConditionBroken" ) );
+        }
+
+        boolean getIt = false;
+
+        if ( isDebug() ) {
+          logDebug( BaseMessages.getString( PKG, "JobEntryFTPDelete.AnalysingFile", filelist[i] ) );
+        }
+
+        try {
+          // First see if the file matches the regular expression!
+          if ( copyprevious ) {
+            if ( list_previous_files.contains( filelist[i] ) ) {
+              getIt = true;
+            }
+          } else {
+            if ( pattern != null ) {
+              Matcher matcher = pattern.matcher( filelist[i] );
+              getIt = matcher.matches();
+            }
+          }
+
+          if ( getIt ) {
+            // Delete file
+            if ( protocol.equals( PROTOCOL_FTP ) ) {
+              ftpclient.delete( filelist[i] );
+            }
+            if ( protocol.equals( PROTOCOL_FTPS ) ) {
+              // System.out.println( "---------------" + filelist[i] );
+              ftpsclient.deleteFile( filelist[i] );
+            } else if ( protocol.equals( PROTOCOL_SFTP ) ) {
+              sftpclient.delete( filelist[i] );
+            } else if ( protocol.equals( PROTOCOL_SSH ) ) {
+              sshclient.rm( sourceFolder + filelist[i] );
+            }
+            if ( isDetailed() ) {
+              logDetailed( "JobEntryFTPDelete.RemotefileDeleted", filelist[i] );
+            }
+            updateDeletedFiles();
+          }
+        } catch ( Exception e ) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**: While wrapping the caught exception into a custom one, information about the caught exception is being lost, including information about the stack trace of the exception.

**Fix**: If the caught exception object does not contain sensitive information, consider passing it as the &#34;rootCause&#34; or inner exception parameter to the constructor of the new exception before throwing the new exception. (Note that not all exception constructors support inner exceptions. Use a wrapper exception that supports inner exceptions.) 
[Learn more](https://www.ibm.com/support/pages/best-practice-catching-and-re-throwing-java-exceptions)